code/datums/quirks/negative.dm

```diff
@@ -424,33 +424,63 @@
 
 /datum/quirk/prosthetic_limb
 	name = "Prosthetic Limb"
-	desc = "An accident caused you to lose one of your limbs. Because of this, you now have a random prosthetic!"
+	desc = "An accident caused you to lose one of your limbs. Because of this, you now have a prosthetic replacement!"
 	icon = "tg-prosthetic-leg"
 	value = -3
 	var/slot_string = "limb"
 	medical_record_text = "During physical examination, patient was found to have a prosthetic limb."
 	hardcore_value = 3
+	var/prosthetic_choice
 
+
+// i've made a real fuckin mess of this frankenstein ass bullshit code i have no idea what i'm doing
 /datum/quirk/prosthetic_limb/add_unique()
-	var/limb_slot = pick(BODY_ZONE_L_ARM, BODY_ZONE_R_ARM, BODY_ZONE_L_LEG, BODY_ZONE_R_LEG)
+	prosthetic_choice = prosthetic_choice || quirk_holder.client?.prefs?.read_preference(/datum/preference/choiced/prosthetic)
 	var/mob/living/carbon/human/human_holder = quirk_holder
 	var/obj/item/bodypart/prosthetic
-	switch(limb_slot)
-		if(BODY_ZONE_L_ARM)
+	switch(prosthetic_choice)
+		if("left arm")
 			prosthetic = new /obj/item/bodypart/l_arm/robot/surplus
 			slot_string = "left arm"
-		if(BODY_ZONE_R_ARM)
+		if("right arm")
 			prosthetic = new /obj/item/bodypart/r_arm/robot/surplus
 			slot_string = "right arm"
-		if(BODY_ZONE_L_LEG)
+		if("left leg")
 			prosthetic = new /obj/item/bodypart/l_leg/robot/surplus
 			slot_string = "left leg"
-		if(BODY_ZONE_R_LEG)
+		if("right leg")
 			prosthetic = new /obj/item/bodypart/r_leg/robot/surplus
 			slot_string = "right leg"
+
+
+
 	human_holder.del_and_replace_bodypart(prosthetic)
 
 /datum/quirk/prosthetic_limb/post_add()
+	//if your preference is set to random, joining at roundstart will cause you to spawn without a limb replacement
+	//for some reason, moving the preference check for that result down here instead of in add_unique prevents this from happening
+	//so if this is breaking anything, uh. sorry! lol!
+	prosthetic_choice = prosthetic_choice || quirk_holder.client?.prefs?.read_preference(/datum/preference/choiced/prosthetic)
+	var/mob/living/carbon/human/human_holder = quirk_holder
+	var/obj/item/bodypart/prosthetic
+	var/random_prosthetic
+	switch(prosthetic_choice)
```

Comment:
`switch` statements are for if you are handling several cases, in this case you are only handling one case
just do `if (prosthetic_choice == "random")` instead.